felix/collector/flowlog/reporter.go

```diff
@@ -154,6 +154,9 @@ func (r *FlowLogReporter) Report(u interface{}) error {
 		return fmt.Errorf("invalid metric update")
 	}
 	log.Debug("Flow Logs Report got Metric Update")
+	if mu.SrcEp != nil && mu.SrcEp.IsHostEndpoint() {
```

Comment:
I think this check is for when reporter == src (unless the SrcEp field doesn't match the eventual source fields in the flow log).  We need to skip if dest is HEP when reporter == dst as well.

Reply:
If we do the same thing for dest reporter, then we will remove all hep logs. I thought it was said to only remove source reporter!

Reply:
The spec is:
- For source reporter, report only flow logs where the source is a pod
- For dest reporter, report only flow logs where the dest is a pod
I've been wording it carefully because it's quite confusing and there are a lot of cases (bear in mind one Felix could be either or both of the source and dest reporter).

Reply:
Ok, essentially the idea is to remove HEP reporter either as source or dest reporter.

Reply:
btw, the proposed changed is basically how it's done in the private repo.